test/cppsuite/test_harness/workload/random_generator.cxx

```diff
@@ -37,25 +42,28 @@ random_generator::instance()
 }
 
 std::string
-random_generator::generate_string(std::size_t length)
+random_generator::generate_string(std::size_t length, characters_type type)
 {
-    std::string random_string;
+    std::string str;
 
-    for (std::size_t i = 0; i < length; ++i)
-        random_string += _characters[_distribution(_generator)];
+    while (str.size() < length)
+        str += std::string(get_characters(type));
 
-    return (random_string);
+    std::shuffle(str.begin(), str.end(), _generator);
+    return (str.substr(0, length));
 }
 
 std::string
```

Comment:
Truely pseudo-random strings would not compress well, and so I think it would be useful to ensure clearly we document the types of pseudo-random strings that can be generated.

Reply:
I am now thinking that might be an erroneous comment that was left behind. @luke-pearson, do you remember the context during which it was created ?

Reply:
Okay the naming of this isn't great its not actually a random string at all we just pick a point on the alphabet string, i.e. 'abcdefghijklmnopqrstuvwxyz' and then chose characters after it in sequence and wrap around if required. I believe that should compress fairly well @jeremythorp what do you think?

Reply:
@luke-pearson and @etienneptl , I agree the naming isn't great, but I don't think it's essential to change the names. Some comments explaining what the class/methods do (with any limitations) should do the trick.

Reply:
Yeah but in regards to compress-ability how well do you think it would compress?

Reply:
I spoke to Jeremy about this directly and we agree that it should work well for compression, obviously shorter string would be problematic but most tests are going to use strings where the pattern would repeat numerous times.

Reply:
@etienneptl if you can think of a better function name I'm happy to change it.

Reply:
@luke-pearson , repeating pattens in long strings should compress well.

Reply:
> @etienneptl if you can think of a better function name I'm happy to change it.

I am ok with `generate_pseudo_random_string` as it reflects what it does. However I changed `generate_string` to `generate_random_string`, I think it is better this way.